src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
Ok now I got confused.
I though about the wrong function.

That actually a good question, I want to move the children, but I have no idea, what should happen, if one NodeId is in a sub tree of the other.
Thats most-likely the use case I had in mind when designing replace.
Move a child higher replacing a parent and dropping the rest in between.